sdk/inc/azure/core/az_result.h

```diff
@@ -99,6 +99,24 @@ enum az_result_core
   /// Dynamic memory allocation request was not successful.
   AZ_ERROR_OUT_OF_MEMORY = _az_RESULT_MAKE_ERROR(_az_FACILITY_CORE_PLATFORM, 1),
 
+  /// Temporary error, resource unavailable temporarily.
```

Comment:
We cannot simply copy everything in POSIX. I assume that other OSs will be different. We should instead have a minimal common set of return values for the PAL that will be determined from the original errors (possible many more) across all PALs.